dev-scheme/racket/racket-8.2-r1.ebuild

```diff
@@ -135,30 +137,27 @@ src_install() {
 }
 
 pkg_preinst() {
-	# If we are merging the same version (NOT revision!)
-	# check if pkg database files exist and do not overwrite them
-	local pvr rel
-	for pvr in ${REPLACING_VERSIONS}; do
-		rel="${pvr%-r*}"
-		if [[ "${rel}" == "${PV}" ]]; then
-			echo "We are installing the same version: ${rel}"
-			local rktd
-			for rktd in "${PKGDB[@]}"; do
-				if [[ -f "${EROOT}/${rktd}" ]]; then
-					einfo "Keeping old file: ${rktd}"
-					mv "${ED}"/${rktd} "${ED}"/${rktd}.bak ||
-						die "failed to create a backup of ${rktd}"
-					cp "${EROOT}"/${rktd} "${ED}"/${rktd} ||
-						die "failed to create a copy of ${rktd}"
-				fi
-			done
-		fi
-	done
+	# If we are merging the same SLOT check if package
+	# database files exist and do not overwrite them
+	if has_version "${CATEGORY}/${PN}:${SLOT}"; then
+		echo "We are installing the same SLOT: ${SLOT}"
+		local rktd
+		for rktd in "${PKGDB[@]}"; do
+			if [[ -f "${EROOT}/${rktd}" ]]; then
+				einfo "Keeping old file: ${rktd}"
+				mv "${ED}"/${rktd} "${ED}"/${rktd}.bak ||
+					die "failed to create a backup of ${rktd}"
+				cp "${EROOT}"/${rktd} "${ED}"/${rktd} ||
+					die "failed to create a copy of ${rktd}"
+			fi
+		done
+	fi
 }
 
 pkg_postinst() {
 	post_X_update
 
+	optfeature "readline editing features in REPL" dev-libs/libedit
```

Comment:
It really dlopens libedit or something? Doesn't link against it?

Reply:
looking into `strace` - racket looks for libreadline in many palces

`openat(AT_FDCWD, "/lib64/libreadline.so.8", O_RDONLY|O_CLOEXEC) = 3`

no, racket does not link against them, it uses FFI to access them

Reply:
wait, libedit or readline?